packages/usa-header/src/index.js

```diff
@@ -188,57 +189,83 @@ const handleEscape = (event) => {
   focusNavButton(event);
 };
 
+/**
+ * Takes clicked link and current url, removes any hash from the link, and compares them.
+ *
+ * @param {HTMLAnchorElement} targetLink - Target link clicked in header.
+ * @returns {boolean} Return `true` if target is same page.
+ */
+const isChildSection = (targetLink) => {
+  const baseURL = new URL(CURRENT_PAGE);
+  const linkDestination = new URL(targetLink.href);
+  if (baseURL.pathname === linkDestination.pathname) {
+    return true;
+  }
+
+  return false;
+};
+
+/**
+ * Sets activeNav to clicked accordion, closes any other open navs.
+ *
+ * @param {HTMLButtonElement} clickedAccordion - Clicked NAV_ACCORDION
+ */
+const updateActiveNav = (clickedAccordion) => {
+  // If another nav is open, close it.
+  if (navActive !== clickedAccordion) {
+    hideActiveNavDropdown();
+  }
+
+  // If no active nav, set navActive to clicked nav.
+  if (!navActive) {
+    navActive = clickedAccordion;
+    toggle(navActive, true);
+  }
+};
+
+/**
+ * Checks if nav link is on the curent page, then closes the mobile nav menu if it is.
+ *
+ * @param {HTMLAnchorElement} navLink - Nav link clicked in header.
+ */
+const handleMobileNav = (navLink) => {
+  if (isChildSection(navLink) && isActive()) {
+    navigation.toggleNav.call(navigation, false);
+  }
+};
+
+/**
+ * Collapse nav accordions when focus leaves primary nav
+ */
+const closeOnFocusOut = (event) => {
+  const nav = event.target.closest(NAV_PRIMARY);
+
+  if (!nav.contains(event.relatedTarget)) {
+    hideActiveNavDropdown();
+  }
+};
+
 navigation = behavior(
   {
     [CLICK]: {
-      [NAV_CONTROL]() {
-        // If another nav is open, close it
-        if (navActive !== this) {
-          hideActiveNavDropdown();
-        }
-        // store a reference to the last clicked nav link element, so we
-        // can hide the dropdown if another element on the page is clicked
-        if (!navActive) {
-          navActive = this;
-          toggle(navActive, true);
-        }
+      [NAV_ACCORDION]() {
+        updateActiveNav(this);
 
         // Do this so the event handler on the body doesn't fire
         return false;
       },
+      [NAV_LINKS]() {
+        handleMobileNav(this);
+      },
       [BODY]: hideActiveNavDropdown,
       [OPENERS]: toggleNav,
       [CLOSERS]: toggleNav,
-      [NAV_LINKS]() {
-        // A navigation link has been clicked! We want to collapse any
-        // hierarchical navigation UI it's a part of, so that the user
-        // can focus on whatever they've just selected.
-
-        // Some navigation links are inside accordions; when they're
-        // clicked, we want to collapse those accordions.
-        const acc = this.closest(accordion.ACCORDION);
-
-        if (acc) {
-          accordion.getButtons(acc).forEach((btn) => accordion.hide(btn));
-        }
-
```

Comment:
## Removed accordion import

- Accordion was required to use the `hide` function on nav link clicks.
- New functions properly collapse accordions without the import.
- Removing the `accordion` import did not cause any regressions.
- Removing `accordion` from `uswds-core` module exports did not cause any regressions not found on current `develop` branch.

Are there any other concerns with this change that I may have missed? Should accordion behaviors be imported by default in case a user does not import the accordion package?

Reply:
I don't believe there is any functional change after my additions. The accordions still collapse as expected! This is still accomplished by the `hideActiveNavDropdown` and `toggle` functions.

https://github.com/uswds/uswds/blob/59192241426774230f6d0a86b6218b1bc3cbf779/packages/usa-header/src/index.js#L166-L173